src/commands/deploy.rs

```diff
@@ -147,11 +154,13 @@ impl DeployCommand {
             self.hippo_username.clone(),
             self.hippo_password.clone(),
         )
-        .await?
-        .token
+        .await
         {
-            Some(t) => t,
-            None => String::from(""),
+            Ok(token_info) => token_info.token.unwrap_or_default(),
+            Err(err) => {
+                let error: LoginHippoError = serde_json::from_str(err.to_string().as_str())?;
+                bail!("Problem login into Hippo: {}", error.detail);
```

Comment:
Minor grammar tweak:

```suggestion
                bail!("Problem logging into Hippo: {}", error.detail);
```

Reply:
You note that if the password is incorrect, the `detail` is blank, so you end up with a trailing colon here (`Problem logging into Hippo: `).  I'd suggest checking for the empty string and dropping the colon (`Problem logging into Hippo`) so it looks a bit tidier in this case.

To avoid this making the `Err` clause so big that it derails the reading flow, you could hive this off into a helper function e.g.

```
Err(err) => bail!(format_login_error(err)?),

// ...

fn format_login_error(err: &Error) -> anyhow::Result<String> {
  let error: LoginHippoError = /* as you have it now */;
  if error.detail.is_empty() {
    // message without colon
  } else {
    // message with colon and detail
  }
}
```

Reply:
The 'detail' field is not blank, and the trailing colon is part of the' detail' field. 

When using an unregistered Hippo username.
> { "title": "Login failed.", "detail": "Login failed: Account does not exist." }

When Hippo password is incorrect.  
> { "title": "Login failed.", "detail": "Login failed:" }

Reply:
```rust
fn format_login_error(err: &anyhow::Error) -> anyhow::Result<String> {
    let error: LoginHippoError = serde_json::from_str(err.to_string().as_str())?;
    if error.detail.ends_with(": ") {
        Ok(format!(
            "Problem logging into Hippo: {}",
            error.detail.replace(": ", ".")
        ))
    } else {
        Ok(format!("Problem logging into Hippo: {}", error.detail))
    }
}
```

When using an unregistered Hippo username.

> Error: Problem login into Hippo: Login failed: Account does not exist.

When Hippo password is incorrect.

> Error: Problem login into Hippo: Login failed.

Reply:
Oops, sorry, my misunderstanding - we should tweak this on the Hippo side.  I like your solution though!

Reply:
(that is to say.... we should do what you suggest even if it's only needed temporarily)